storage/gcs/src/integration-test/java/io/aiven/kafka/tieredstorage/storage/gcs/GcsStorageMetricsTest.java

```diff
@@ -0,0 +1,143 @@
+import io.aiven.kafka.tieredstorage.storage.StorageBackendException;
+
+import com.google.cloud.NoCredentials;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+@Testcontainers
+public class GcsStorageMetricsTest {
+    private static final int RESUMABLE_UPLOAD_CHUNK_SIZE = 256 * 1024;
+
+    @Container
+    static final FakeGcsServerContainer GCS_SERVER = new FakeGcsServerContainer();
+
+    static final MBeanServer MBEAN_SERVER = ManagementFactory.getPlatformMBeanServer();
+    static GcsStorage storage;
+    static Storage storageClient;
+    private String bucketName;
+
+    @BeforeAll
+    static void setUpClass() throws Exception {
+        storageClient = StorageOptions.newBuilder()
+            .setCredentials(NoCredentials.getInstance())
+            .setHost(GCS_SERVER.getURL())
+            .setProjectId("test-project")
+            .build()
+            .getService();
+    }
+
+    @BeforeEach
+    void setUp(final TestInfo testInfo) {
+        bucketName = testInfo.getDisplayName()
+            .toLowerCase()
+            .replace("(", "")
+            .replace(")", "");
+        while (bucketName.length() < 3) {
+            bucketName += bucketName;
+        }
+        storageClient.create(BucketInfo.newBuilder(bucketName).build());
+
+        storage = new GcsStorage();
+        final Map<String, Object> configs = Map.of(
+            "gcs.bucket.name", bucketName,
+            "gcs.endpoint.url", GCS_SERVER.getURL(),
+            "gcs.resumable.upload.chunk.size", Integer.toString(RESUMABLE_UPLOAD_CHUNK_SIZE),
+            "gcs.credentials.default", "false"
+        );
+        storage.configure(configs);
+    }
+
+    @Test
+    void metricsShouldBeReported() throws StorageBackendException, IOException, JMException {
+        final byte[] data = new byte[RESUMABLE_UPLOAD_CHUNK_SIZE + 1];
+
+        final String key = "x";
+
+        storage.upload(new ByteArrayInputStream(data), key);
+        try (final InputStream fetch = storage.fetch(key)) {
+            fetch.readAllBytes();
+        }
+        try (final InputStream fetch = storage.fetch(key, BytesRange.of(0, 1))) {
+            fetch.readAllBytes();
+        }
+        storage.delete(key);
+
+        final ObjectName gcsMetricsObjectName =
+            ObjectName.getInstance("aiven.kafka.server.tieredstorage.gcs:type=gcs-metrics");
+        assertThat((double) MBEAN_SERVER.getAttribute(
```

Comment:
Should we keep using `.asInstanceOf(DOUBLE)` like in other places?

Reply:
Yes, changed